actors/migration/top.go

```diff
@@ -118,26 +114,40 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, stateRootIn cid
 	if err != nil {
 		return cid.Undef, err
 	}
+	// Setup input and output state tree helpers
 	adtStore := adt.WrapStore(ctx, store)
 	actorsIn, err := states.AsTreeTop(adtStore, stateRootInTweaked)
 	if err != nil {
 		return cid.Undef, err
 	}
-	var p phoenix
-	if err := p.load(ctx, actorsIn); err != nil {
-		return cid.Undef, err
-	}
-
 	stateRootOut, err := adt.MakeEmptyMap(adtStore).Root()
 	if err != nil {
 		return cid.Undef, err
 	}
 	actorsOut, err := states.AsTreeTop(adtStore, stateRootOut)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	// Extra setup
+	// miner
+	var p phoenix
+	if err := p.load(ctx, actorsIn); err != nil {
+		return cid.Undef, err
+	}
+	// power
+	pm := migrations[builtin0.StoragePowerActorCodeID].StateMigration.(*powerMigrator)
+	pm.actorsIn = actorsIn
 
 	// Iterate all actors in old state root
 	// Set new state root actors as we go
 	err = actorsIn.ForEach(ctx, func(addr address.Address, actorIn *states.Actor) error {
 		migration := migrations[actorIn.Code]
+
+		// This will be migrated at the end
+		if actorIn.Code == builtin0.VerifiedRegistryActorCodeID {
```

Comment:
You can just check for the singleton address (`t080`), and if the address is not that, assert it's not a verified registry.